codemods/styled-components-to-ucl.ts

```diff
@@ -162,3 +216,30 @@ export function transformStyledCompoentsToUCL(
     styledImport.remove();
   }
 }
+
+type ASTTypes = typeof types.namedTypes;
+
+function walkUpToFind<T extends keyof ASTTypes>(
+  node: any,
+  j: JSCodeshift,
+  type: T,
+  // ): ASTTypes[T] | null {
```

Comment:
tried to do good types.. couldnt get it right

Reply:
This seems fine, but just incase you were not aware there's also `closest` and `find`
https://github.com/facebook/jscodeshift/wiki/jscodeshift-Documentation#closest

Reply:
from my testing it seemed like closest and find went down the tree and not up, and i need to walk up. is that right?